Upgrade 1.x precommit and cat vrs submodule

larrybabb opened the pull request.

larrybabb committed:
- updated cat-vrs submodule
- correct schema references to upstream gks 2025-02 snapshot branches
- add pre-commit hooks to clean up erradic whitespace in files.

larrybabb committed:
fixed tests to work with upgraded submodule changes (mappableConcept.primaryCoding)

larrybabb committed:
missed whitespace fix

larrybabb committed:
Merge branch '1.x' into upgrade-1.x-precommit-and-cat-vrs-submodule

larrybabb committed:
fix for specifiedBy in VariantPathogenicityFunctionalImpactEvidenceLine

mbrush committed:
Added mappable concept value examples/guidance per #262 (#280)

Added  examples of terminology systems / terms to consider  as values for mode of inheritance qualifier, and allele origin qualifier

mbrush commented on schema/va-spec/ccv-2022/oncogenicity-functional-impact-evidence-line-profile-source.yaml:
```diff
@@ -29,46 +29,52 @@ $defs:
             as evidence for or against the assessed variant's oncogenicity.
           properties:
-            primaryCode:
+            primaryCoding:
```
@larrybabb I think you'll want to update this to match what we decided for the analogous property in the PathogenicityFunctionalImpactEvidenceLine profile [here](https://github.com/ga4gh/va-spec/blob/da285fca045dcdee3d1f5e9a7713cd1270b2319b/schema/va-spec/acmg-2015/pathogenicity-functional-impact-evidence-line-profile-source.yaml#L45) - where we decided the `specifiedBy` property needs to take a Method, and that the best we can do here for now is constrain the value of the `Method.isReportedIn` attribute here to capture the interpretation guideline document.  

For Pathogenicity  we used the ACMG 2015 paper.  For oncogenicity, I don't think there is a formal publication/pmid - so we will need another document to reference - maybe [this one](https://clinicalgenome.org/docs/standards-for-the-classification-of-pathogenicity-of-somatic-variants-in-cancer-oncogenicity-joint-recommendations-of-clinical/)?

larrybabb committed:
VPFuncImpEvLine requirement for `reportedIn`

larrybabb committed:
Issue #250 Generalize ancillaryResults & qualityMeasures

mbrush committed:
Update va-core-source.yaml (#285)

- Moving ancillaryResults and qualityMeasures into the core StudyResult class (they were previously defined in the CAF profile) - per decision documented in #250
- added descriptions of these attributes as well, as they were missing before.

mbrush committed:
Change `Contribution.date` datatype to datetime (#277)

Addresses  #240

mbrush committed:
Clarify when geneContextQualifier should be provided (#278)

* Clarify when geneContextQualifier should be provided
* Added a comment with implementation guidance on this point.

mbrush committed:
Updating specification for Condition, TraitSet and Therapy Group based on #252 and #253 (#279)

* Updating specification for TraitSet and Therapy Group based on #252
* TraitSet --> ConditionSet - as everywhere else in the spec (including the description of elements in this class) we call these conditions, not traits.
* added text to comment on Condition class to additionally address #253 (how to report that no condition is provided).

dependabot[bot] committed:
Bump jinja2 from 3.1.5 to 3.1.6 in /docs/source (#282)

Bumps [jinja2](https://github.com/pallets/jinja) from 3.1.5 to 3.1.6.

mbrush committed:
Update pathogenicity-functional-impact-evidence-line-profile-source.yaml (#286)

- updated language used to describe the evidenceOutcome attribute, which were referring to a specific 'specifiedBy' criterion code, but we no longer capture a specific code in the 'specifiedBy; field. This fix is essential.
- also updated the description of this attribute and 'strengthOfEvidenceProvided' to include language explaining the nesting of the enumerations defined for these attributes. These fixes are optional in my mind - but simply add some clarity around how.where the enumeration binding acts.

mbrush committed:
Update oncogenicity-functional-impact-evidence-line-profile-source.yaml (#287)

- updated language used to describe the evidenceOutcome attribute, which were referring to a specific 'specifiedBy' criterion code, but we no longer capture a specific code in the 'specifiedBy; field. This fix is essential.
- also updated the description of this attribute and 'strengthOfEvidenceProvided' to include language explaining the nesting of the enumerations defined for these attributes. These fixes are optional in my mind - but simply add some clarity around how.where the enumeration binding acts.

larrybabb committed:
- fix incorrect references to 1.0.0-snapshot release for va-spec related classes
- pre-commit apply to fix whitespacing
- update test cases for conditionset/therapygroup
